groups/reconcile.go

```diff
@@ -0,0 +1,392 @@
+
+			l, err := srv.Members.List(g.Email).Do()
+			if err != nil {
+				log.Fatalf("Unable to retrieve members in group : %v", err)
+			}
+
+			if len(l.Members) == 0 {
+				fmt.Print("No members found in group.\n")
+			} else {
+				for _, m := range l.Members {
+					fmt.Printf(">>> %s (%s)\n", m.Email, m.Role)
+				}
+			}
+			fmt.Printf("\n")
+		}
+	}
+}
+
+func createGroupIfNecessary(srv *admin.Service, groupEmailId string, description string) {
+	_, err := srv.Groups.Get(groupEmailId).Do()
+	if err != nil {
+		switch err := err.(type) {
+		case *googleapi.Error:
+			if err.Code == http.StatusNotFound {
+				if !config.DryRun {
+					fmt.Printf("Trying to create group: %s\n", groupEmailId)
+					g4, err := srv.Groups.Insert(&admin.Group{
+						Email:       groupEmailId,
+						Name:        description,
+						Description: "Kubernetes wg-k8s-infra test group #2",
+					}).Do()
+					if err != nil {
+						log.Fatalf("Unable to add new group %s: %v", groupEmailId, err)
+					} else {
+						fmt.Printf("> Successfully created group %s\n", g4.Email)
+					}
+				} else {
+					fmt.Printf("dry-run : skipping creation of group %s\n", groupEmailId)
+				}
+			} else {
+				log.Fatalf("Unable to fetch group: %#v", err)
+			}
+		default:
+			log.Fatalf("Unable to fetch group: %#v", err.Error())
+		}
+	}
+}
+
+func deleteGroupsIfNecessary(service *admin.Service) {
+	g, err := service.Groups.List().Customer("my_customer").OrderBy("email").Do()
+	if err != nil {
+		log.Fatalf("Unable to retrieve users in domain: %v", err)
+	}
+	if len(g.Groups) == 0 {
+		fmt.Print("No groups found.\n")
+	} else {
+		for _, g := range g.Groups {
+			// Don't touch existing mailing lists, we should
+			// always prefix with "k8s-infra-"
+			if !strings.HasPrefix(g.Email, "k8s-infra-") {
+				continue
+			}
+			found := false
+			for _, g2 := range groupsConfig.Groups {
+				if g2.EmailId == g.Email {
+					found = true
+					break
+				}
+			}
+			if !found {
+				if !config.DryRun {
+					err := service.Groups.Delete(g.Email).Do()
+					if err != nil {
+						log.Fatalf("Unable to remove group %s : %v", g.Email, err)
+					} else {
+						fmt.Printf("Removing group %s\n", g.Email)
+					}
+				} else {
+					fmt.Printf("dry-run : Skipping removing group %s\n", g.Email)
+				}
+			}
+		}
+	}
+}
+
+func updateGroupSettingsToAllowExternalMembers(srv *groupssettings.Service, groupEmailId string) {
+	g2, err := srv.Groups.Get(groupEmailId).Do()
+	if err != nil {
+		switch err := err.(type) {
+		case *googleapi.Error:
+			if err.Code == http.StatusNotFound {
+				fmt.Printf("skipping updating group settings as group %s has not yet been created\n", groupEmailId)
+				return
+			} else {
+				log.Fatalf("Unable to retrieve group info for group %s: %v", groupEmailId, err)
+			}
+		default:
+			log.Fatalf("Unable to retrieve group info for group %s: %v", groupEmailId, err)
+		}
+	}
+	if g2.AllowExternalMembers != "true" {
+		if !config.DryRun {
```

Comment:
Can you flip this to be `if config.DryRun` (use a positive boolean for the `if`)?